### ISSUE TITLE
Feature request: Option for less obtrusive tool icon (50x50 area instead of 100x100)

### ISSUE THREAD

**Issue by sztanj** [mb1986/rm-hacks#88]
I got used to the ddvk hacks which packed the tool icon into roughly 50x50 pixels in the very corner of the screen. The rm-hacks claim the same icon area as the native firmware, which seems to be 100x100 pixels or four times larger than ddvk. 

From ddvk I am used to using the top left corner of the page for dates, etc, and I don't like the obtrusive icon and its larger hitbox. Would it be possible to add an option for less obtrusive tool icon?

Below are screenshots from ddvk and from rm-hacks 0.0.7 along with a mockup that keeps the large color swatch, but packs it into a smaller 50x50 px area like ddvk.

Current rm-hacks marker icon 100x100:
![231211_rm-hacks_tool-icon_100x100](https://github.com/mb1986/rm-hacks/assets/153586554/8dab032a-a735-4a46-9162-26bc76f08148)

Mockup of same marker icon packed into 50x50:
![231211_rm-hacks_tool-icon_50x50_packed](https://github.com/mb1986/rm-hacks/assets/153586554/74a81ba4-0345-4db5-b987-602afd73416f)

ddvk screenshot with eraser icon:
![231123_2 15ddvk_Toolbar closed](https://github.com/mb1986/rm-hacks/assets/153586554/eff93f89-a548-4c2f-a733-589f61e01175)

rm-hacks screenshot with eraser icon:
![231211_3 8x_rm-hacks0 0 7_Toolbar closed_eraser](https://github.com/mb1986/rm-hacks/assets/153586554/bc6aeb8d-d236-4a56-b119-60326bf7de7d)

Please consider this.
But regardless, thank you for all the work you have done!

**Comment by PepikVaio** [comment]
I, on the other hand, like that it's consistent and if you have a PDF that matches the rm panel interface, the 100x100 icon is better because it doesn't break the look.

If you prefer the icon on the left, move the panel to the left.

**Comment by Tanishama** [comment]
I think it is a great idea. Coming from Ddvk I appreciated the small size of that icon. It is ok the way it is now, but the more space the better. Maybe it could be optional.

**Comment by rudy0ne** [comment]
I would not want that enhancement to come with a reduction of the gesture activation area. 

Also, the circular color swatch is often useless on its own and would still benefit from an abbreviation (#71) of the color. But perhaps something like in this screenshot could work:

![Screen Shot 2023-12-12 at 9 21 02 AM](https://github.com/mb1986/rm-hacks/assets/22302732/cf9bc3a9-4d2b-495d-8bad-622db17115b9)

**Comment by martinhess-at** [comment]
> I got used to the ddvk hacks which packed the tool icon into roughly 50x50 pixels in the very corner of the screen. The rm-hacks claim the same icon area as the native firmware, which seems to be 100x100 pixels or four times larger than ddvk.
> 
> From ddvk I am used to using the top left corner of the page for dates, etc, and I don't like the obtrusive icon and its larger hitbox. Would it be possible to add an option for less obtrusive tool icon?
> 
> Below are screenshots from ddvk and from rm-hacks 0.0.7 along with a mockup that keeps the large color swatch, but packs it into a smaller 50x50 px area like ddvk.
> 
> Current rm-hacks marker icon 100x100: ![231211_rm-hacks_tool-icon_100x100](https://private-user-images.githubusercontent.com/153586554/289775210-8dab032a-a735-4a46-9162-26bc76f08148.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMxNTkwNjQsIm5iZiI6MTcwMzE1ODc2NCwicGF0aCI6Ii8xNTM1ODY1NTQvMjg5Nzc1MjEwLThkYWIwMzJhLWE3MzUtNGE0Ni05MTYyLTI2YmM3NmYwODE0OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIyMVQxMTM5MjRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jNWNkNGFhY2RmNTg0NGJkZDNkMzk2MjQwZGIzMGNkMGZkYWM1NmYzMWQ5MWM1OGI3M2NlZWYwMWFjNmNkM2IxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.1LXbwZPbGFQb_QvX6bUvdLr1--mz_KwOHSVMVS0sLi0)
> 
> Mockup of same marker icon packed into 50x50: ![231211_rm-hacks_tool-icon_50x50_packed](https://private-user-images.githubusercontent.com/153586554/289775214-74a81ba4-0345-4db5-b987-602afd73416f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMxNTkwNjQsIm5iZiI6MTcwMzE1ODc2NCwicGF0aCI6Ii8xNTM1ODY1NTQvMjg5Nzc1MjE0LTc0YTgxYmE0LTAzNDUtNGRiNS1iOTg3LTYwMmFmZDczNDE2Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIyMVQxMTM5MjRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iNWEwMzNhYmVmOGNlYzg3MTc5NTBmZTI0YzcyZTQ5Yzk0YjU1MDAzNWMzZmY0NjhjODQ5YTEwNGIwODkxNDI5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.CGBfzuPDYgSHGpSkDGiR3DrbQNaBqIHP3B8dCqYhI04)
> 
> ddvk screenshot with eraser icon: ![231123_2 15ddvk_Toolbar closed](https://private-user-images.githubusercontent.com/153586554/289775487-eff93f89-a548-4c2f-a733-589f61e01175.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMxNTkwNjQsIm5iZiI6MTcwMzE1ODc2NCwicGF0aCI6Ii8xNTM1ODY1NTQvMjg5Nzc1NDg3LWVmZjkzZjg5LWE1NDgtNGMyZi1hNzMzLTU4OWY2MWUwMTE3NS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIyMVQxMTM5MjRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02NDNhMjA3ZjYxYzAyMWViZjhkNWYzMjY2OWVhMjkzMWUyNzhhYzE2NGU0Y2Q1M2MyODkwYjU3YmNjYmI5ZGE1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.tKP--nJX2PmWENoHf7_a8aPtCir2MDCccHMh8oeHEIU)
> 
> rm-hacks screenshot with eraser icon: ![231211_3 8x_rm-hacks0 0 7_Toolbar closed_eraser](https://private-user-images.githubusercontent.com/153586554/289775208-bc6aeb8d-d236-4a56-b119-60326bf7de7d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMxNTkwNjQsIm5iZiI6MTcwMzE1ODc2NCwicGF0aCI6Ii8xNTM1ODY1NTQvMjg5Nzc1MjA4LWJjNmFlYjhkLWQyMzYtNGE1Ni1iMTE5LTYwMzI2YmY3ZGU3ZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIyMVQxMTM5MjRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yNzdiNWQxNmJhZmM1YzYyOWU3NDkwNWUwODYwZjlhZmMwMWVmZTkzYTcyMmU1NWVjNDQ0NDg1ODM3OTUzZmJkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.QzaNx-Mg7IjsbvXRliP0sWUSXIf1GBF1Hf62orl-FgA)
> 
> Please consider this. But regardless, thank you for all the work you have done!


hi there! i came the same way from 2.14. with ddvk and then was stuck with no hack at 3.8.3.1986.
since 0.0.7. is now open for the version i'm astonished how different it is to ddvks now. and i'm missing some things. one of which is the small icon on the top of course. i think rudys and sztanjis proposals would be a big improvement.. 
have a nice day, thanks
martin

**Comment by sztanj** [comment]
I agree with the suggestions to make this optional and to include indication of the tool color if possible.

> I would not want that enhancement to come with a reduction of the gesture activation area.

I also agree with this comment from @rudy0ne. I like the fingertip swipe gestures in the upper left corner and find them easier to trigger in rm-hacks than they were in ddvk so I don't want to change their implementation.

In mentioning the icon area's "hitbox" in my original post, I more meant that I am unable to start a pen stroke within the 100x100 icon area, even though much of it is empty space. With ddvk I could start pen strokes in the opened-up parts of the icon area, and that gives more freedom to write in the upper left of the page.

**Comment by PepikVaio** [comment]
@mb1986 You're the best.
Thank you for opening the suggestion to change the look of the bookmarks, it would be nice.

**Comment by mb1986** [comment]
I am just messing with sizes of the toolbar itself. The original width of the toolbar is 104. I think reducing it to 75% which is 78 would be fine. Smaller value may be to small in my opinion.

On the other hand I'll try to reduce the brush icon to about 50.

I think I'll add two different configuration entries to enable/disable them separately.

What do you think?

**Comment by PepikVaio** [comment]
I think that would be about right.
I don't know how wide the supernote is, but it's about 9 mm.

75% reMarkable would be about 8.25 mm.
That makes sense to me.

**Comment by PepikVaio** [comment]
Send a test of the code and we'll try both 50px and 75px variants

**Comment by mb1986** [comment]
It cannot be as small as 54... 😭

![IMG20240109203837.jpg](https://github.com/mb1986/rm-hacks/assets/1386931/1c0e205a-0cb8-4241-92e1-3133dd0d9f11)

**Comment by PepikVaio** [comment]
It is 54 px?
And icon is to smaller too?

**Comment by mb1986** [comment]
The icon is fine, but the pen size or color is not readable. I'll reduce it to 75% of the original size, and then you can test it.

**Comment by PepikVaio** [comment]
OK

**Comment by mb1986** [comment]
This one is ready for testing:
```sh
sh -c "$(wget https://raw.githubusercontent.com/mb1986/rm-hacks/main/install.sh -O-)" _ patch 0.0.9-alpha.3
```

**Comment by sztanj** [comment]
I like it! Thank you very much! 
To me this satisfies my original request.

- The entire toolbar is much less obtrusive at 75%, which I like.
- When the toolbar is collapsed, the tool icon is less obtrusive and packed much closer into the corner.
- The "hitbox" of the icon (where I cannot start a new stroke) is much, much smaller than before. Previously it was 100% of the toolbar width, but now it is less than the 75% toolbar width. The new "hitbox" is only the region immediately under the icon graphic and the text labels. This will let me write and draw closer to the corner again.

![240111_rm-hacks_tool-icon_0 0 9 a_packed_hitboxes]
(https://github.com/mb1986/rm-hacks/assets/153586554/7e96a965-56a0-43a9-8119-ec508afe58b0)
↑ Old hitbox in red, new smaller hitboxes in green

**Comment by dreadful-brain** [comment]
I have just tested it and all I can say is: AWESOME. 
The menu looks so much better than before. You are amazing!

**Comment by martinhess-at** [comment]
Congrats! The new menu is supersweet ;)

**Comment by mb1986** [comment]
I am happy you like it :)

**Comment by PepikVaio** [comment]
I mean, big change.
What is the width in px please?
Did you make a percentage change or did you set the size in px?
I have a template linked to it and it matches my bar (not now), so I want to know how to adjust.

Appearance:
It's different, at first I thought it was weird (unusual).
Now it's a bit better :)
And tomorrow it will be a blast and I won't be able to be without it :) And once I switch back, I'll be wondering what the heck???

Edit:
As soon as I hide the sidebar, the ball slides up.
I hated this with DDVK :(
It's distracting because it makes a certain animation and ugly.
Ideally, the ball should still be docked in the same place.

Workaround:
Once the ball stays in place the pen icon (if enabled) will be lower (now travels up) and there will be no contact between the pen icon and the selected color.
But I would also put some pixels there so they don't touch the edge of the screen. Like the clock (if it's displayed), it seems to me to have a few pixels there.

**Comment by PepikVaio** [comment]
One more problem:
The information at the bottom is now wider than the panel itself and it doesn't fit the eye.

![IMG_3251](https://github.com/mb1986/rm-hacks/assets/36114753/fd66e076-242e-4ced-bac3-d6a2a23c884e)

**Comment by mb1986** [comment]
> What is the width in px please?
> Did you make a percentage change or did you set the size in px?

I've reduced `104` to `78` (75%).

**Comment by PepikVaio** [comment]
I've now realised that when the panel is closed, that's when the ball is in the right place.
Equally aligned from the left (or right) and from the top.

This cancels out the problem with touching the pen icon and text (color).

I mean, can the panel move up when open?
When the panel is open, the ball scrolls down.

Thank you

**Comment by Kelvin88** [comment]
Slim toolbar and icon works. Like that a lot.

**Comment by PepikVaio** [comment]
Hi, please let me know if the moving icon when opening and closing the panel will be fixed?
Or does it make any sense?

Thank you, I am sending a video of testing.

https://github.com/mb1986/rm-hacks/assets/36114753/f8eb5e02-b116-44bd-a908-b4ea99677f49

**Comment by martinhess-at** [comment]
Hi Josef, how do you manage to see the ball in a document, when the menu is hidden? Normally there is a tool visible, no? Its just when you make a new notebook? But i see it in the video testing. Me curious how many requests mb1986 will be wanting and able to solve. But anyway, very nice work for us! And for the use of the rM... ;) Thanks & Regards Martin

**Comment by PepikVaio** [comment]
You can choose to see the tool (hack) or turn it off and see the ball (original).

**Comment by martinhess-at** [comment]
Ahh, i see!

**Comment by PepikVaio** [comment]
> Hi, please let me know if the moving icon when opening and closing the panel will be fixed? Or does it make any sense?
> 
> Thank you, I am sending a video of testing.
> 
>  IMG_3268.mov

Now I know why the ball is moving.
I figured it out when I took a moment to view the clock.
When the panel closes, the sphere is placed where it's supposed to be.
But when the panel is opened, it moves down like in the video because that's where the clock is kept.
But I don't have the clock active, so I don't need the space.

I get it now, but can it be fixed?
When the clock is not displayed, the panel will not travel down.

**Comment by PepikVaio** [comment]
> One more problem: The information at the bottom is now wider than the panel itself and it doesn't fit the eye.
> 
> ![IMG_3251](https://private-user-images.githubusercontent.com/36114753/296040295-fd66e076-242e-4ced-bac3-d6a2a23c884e.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYxMjM1NTIsIm5iZiI6MTcwNjEyMzI1MiwicGF0aCI6Ii8zNjExNDc1My8yOTYwNDAyOTUtZmQ2NmUwNzYtMjQyZS00Y2VkLWJhYzMtZDZhMmEyM2M4ODRlLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTI0VDE5MDczMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA3YmJjYzJjMTBlNzU5YTc4NzAyMzRmNDk3MWYwN2IwYzZiZGM3N2I2ODUxZWY3ZjEzYzZkMjkyMDZiMWQ1ZjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.3AIdTBbvqWtNb_a7HTRRG56_n3iEg9yWX8O8je8Rn1w)

Here I would also solve it so that the wifi information etc has a separate place on the panel, the tribute does not fit when shrinking as when the panel is the original wide.

**Comment by knox-dawson** [comment]
I wonder if this might be an answer (plug for my wish list issue):
https://github.com/mb1986/rm-hacks/issues/109
